gammapy/utils/random/inverse_cdf.py

```diff
@@ -77,3 +83,125 @@ def sample(self, size):
 
         index = index + self.random_state.uniform(low=-0.5, high=0.5, size=index.shape)
         return index
+
+
+class MapEventSampler:
+    """Map event sampler.
+
+    Parameters
+    ----------
+
+    npred_map : `~gammapy.maps.Map`
+            Predicted number of counts map.
+    random_state : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}
+            Defines random number generator initialisation.
+            Passed to `~gammapy.utils.random.get_random_state`.
+    lc : `~gammapy.time.models.LightCurveTableModel`
+            Input light-curve model of the source, given with columns labelled
+            as "time" (in seconds) and "normalization" (arbitrary units): the bin time
+            HAS to be costant.
+    phase_lc : `~gammapy.time.models.PhaseCurveTableModel`
+            Input phase-curve model of the source, given with columns labelled
+            as "Phase" and "normalization" (arbitrary units): the bin time
+            HAS to be costant.
+    tmin : float
```

Comment:
I think we should make `t_min` and `t_max` an `astropy.time.Time` object. This allows users to attach units as well as handling different reference times.

Reply:
I've changed the code as that tmin and tmax may be either `astropy.units.Quantity` or `astropy.time.TIME` objects. In the second case, the times are then all referred to a reference time, hardcoded in the MapEventSampler class. These are the changes:

`from astropy.time import Time

 tmin =  ...
 tmax = ...

 if isinstance(tmin,Time) and isinstance(tmax,Time):
              t_ref = Time('2019-01-01 00:00:00', scale='utc') #hardcoded reference epoch
              self.tmin = ((tmin.decimalyear - t_ref.decimalyear) * u.year).to(u.s)
              self.tmax = ((tmax.decimalyear - t_ref.decimalyear) * u.year).to(u.s)
        
 else:
              self.tmin = tmin.to(u.s)
              self.tmax = tmax.to(u.s)
`